lib/vyasa/sangh/sheaf.ex

```diff
@@ -42,11 +42,24 @@ defmodule Vyasa.Sangh.Sheaf do
   @doc false
   def changeset(%Sheaf{} = sheaf, attrs) do
     sheaf
-    |> cast(attrs, [:id, :body, :active, :session_id, :signature, :traits])
+    |> cast(attrs, [
+      :id,
+      :body,
+      :active,
+      :session_id,
+      :signature,
+      :traits,
+      :updated_at,
+      :inserted_at
+    ])
     |> cast_path(attrs)
     |> assoc_marks(attrs)
     |> validate_required([:id, :session_id, :path])
-    |> validate_include_subset(:traits, ["personal", "draft", "publish"])
+    |> validate_include_subset(:traits, ["personal", "draft", "published"])
+
+    # QQ: @ks0m1c in my mind, i see private vs public and draft vs published as two distinct dimensions
+    # and we'd want to filter by these dimensions separately. Therefore, I wonder if it's better to NOT keep
+    # them all as string identifiers within a list.
```

Comment:
yeah extending the B)

A Personal Sangh Session is ever present, always present for them to make private trails as expressed as sheafs or tap unto learnings from private trails as expressed by marks in sheafs

One question when does a personal sangh session become I public when sangh_id becomes parameterised in url params primed by [sharing] when user clicks this share button the personal has become published to an audience of one if they share to a close friend or many depending on their intent.

If they receive a link to anothers public sangh_id they can tap unto marks of relevance and by proxy sheafs (if people decide to jump into their marks) from their personal world but here they do not publish their entire lineage of sheafs just one which they had quoted and binded that is relevant

What are some questions that only testing will reveal do people prefer on sharing to fork their private session and create another public session from that 
OR
sharing creates  a new blank private session while publishing their entire set of sheafs
and how much control do they want in what they share to this newly published session do they want granularity in which sheafs they want to publish for e.g

These are questions but I say not critical, it is determined much by what behaviour feels coherent and flows better based on our observations of user behaviour

Reply:
> sangh_id becomes parameterised in url params primed by [sharing]

we can stick to using sheafs since sheafs have sangh_ids associated with it


> One question when does a personal sangh session become I public when sangh_id becomes parameterised in url params primed by [sharing] when user clicks this share button the personal has become published to an audience of one if they share to a close friend or many depending on their intent.

^ bubbling analogy  -- we have to deliberate more on this but just touch the surface for this release



> they had quoted a
analogy: there's a discussion going on, and you already have a point you wanna make and you contextualise it to the flow of the discussion
in this case you would want to FORK your mark/sheaf then edit and publish it, it's not a direct edit because you wanna keep your original mark/sheaf
==> this is what the quoting flow is for -- quoting == create binding on your mark/sheaf ==> group it up as a sheaf and publish in the public discussion place


----


LEANEST SCOPE FOR BUBBLING: 

as soon as you do a share, your personal sangha session is immediately public, and you, the unique user identity is issued a new personal sangha session



similar to excalidraw


--- 

USER MODEL 
- unique identifier for a user? no need becasue the context is sharing with trusted parties 
- so it's a "i chope this signature for this particular sangh session" 
- skip the email-based auth stuff for now

Reply:
uuid by default gen for anon
name rich and flexible across sessions
email only for confirmation and recovery (as needed)